packages/2d/src/components/CodeBlock.ts

```diff
@@ -47,14 +50,44 @@ export interface CodeModification {
 }
 
 export class CodeBlock extends Shape {
-  private static initialized = createComputedAsync(
-    () => ready().then(() => true),
-    false,
-  );
-
   @initial('tsx')
   @signal()
-  public declare readonly language: SimpleSignal<string, this>;
+  public declare readonly language: SimpleSignal<Lang, this>;
+
+  @initial(undefined)
+  @signal()
+  public declare readonly stockTheme: SimpleSignal<Theme | undefined, this>;
+
+  protected static loadedLanguages = new Map<Lang, boolean>();
+  protected static loadedThemes = new Map<Theme, boolean>();
+
+  @computed()
+  protected isReady() {
+    const language = this.language();
+    const theme = this.stockTheme();
+    if (
+      !CodeBlock.loadedLanguages.has(language) ||
+      (theme != null && !CodeBlock.loadedThemes.has(theme))
+    ) {
+      DependencyContext.collectPromise(
+        ready({
+          languages: [language],
+          themes: theme != null ? [theme] : [],
+        }).then(() => {
+          CodeBlock.loadedLanguages.set(language, true);
+          if (theme != null) {
+            CodeBlock.loadedThemes.set(theme, true);
+          }
+        }),
+      );
+      return false;
+    }
+
+    return (
+      CodeBlock.loadedLanguages.get(language) &&
+      (theme == null || CodeBlock.loadedThemes.get(theme))
```

Comment:
Couldn't it be:
```suggestion
      CodeBlock.loadedLanguages.has(language) &&
      (theme == null || CodeBlock.loadedThemes.has(theme))
```
?

Reply:
I updated the code.